pd/src/info/light_wallet.rs

```diff
@@ -0,0 +1,160 @@
+use std::pin::Pin;
+
+use async_stream::try_stream;
+use futures::stream::{StreamExt, TryStreamExt};
+use penumbra_proto::{
+    chain::{ChainParams, CompactBlock, KnownAssets},
+    light_wallet::{
+        light_wallet_server::LightWallet, AssetListRequest, ChainParamsRequest,
+        CompactBlockRangeRequest, ValidatorInfoRequest,
+    },
+    stake::ValidatorInfo,
+    Protobuf,
+};
+use tonic::Status;
+use tracing::instrument;
+
+// TODO(hdevalence): this still doesn't work, giving up for now
+// We need to use the tracing-futures version of Instrument,
+// because we want to instrument a Stream, and the Stream trait
+// isn't in std, and the tracing::Instrument trait only works with
+// (stable) std types.
+// use tracing_futures::Instrument;
```

Comment:
Supposedly there's a way to use the `Instrument` trait to attach a tracing span to a `Stream`, but I couldn't get it to work, and didn't think it was worth spending more time on right now.